app/src/main/java/io/livekit/android/example/voiceassistant/MainActivity.kt

```diff
@@ -38,116 +38,124 @@ import io.livekit.android.example.voiceassistant.ui.theme.LiveKitVoiceAssistantE
 import io.livekit.android.room.track.Track
 import io.livekit.android.util.LoggingLevel
 
-// Replace these values with your url and generated token.
-const val wsURL = ""
-const val token =
-    ""
-
 class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         LiveKit.loggingLevel = LoggingLevel.DEBUG
         requireNeededPermissions {
-            setContent {
-                LiveKitVoiceAssistantExampleTheme {
-                    VoiceAssistant(modifier = Modifier.fillMaxSize())
+            requireToken { url, token ->
+                setContent {
+                    LiveKitVoiceAssistantExampleTheme {
+                        VoiceAssistant(
+                            url,
+                            token,
+                            modifier = Modifier.fillMaxSize()
+                        )
+                    }
                 }
```

Comment:
@davidliu I left it in this "blocking" style because `okhttp` doesn't expose a `suspend` interface as far as I could tell. Feel free to refactor if I'm missing something!